components/uniset/src/uniset.rs

```diff
@@ -9,8 +9,10 @@ const BMP_MAX: u32 = 0xFFFF;
 ///
 /// Provides exposure to membership functions and constructors from serialized UnicodeSets
 /// and predefined ranges.
-#[derive(Debug, PartialEq)]
+#[derive(Debug, PartialEq, Hash, Eq)]
 pub struct UnicodeSet {
+    // TODO: need advice - how should we remove Hash and Eq from UnicodeSet unless we need it?
```

Comment:
Using a UnicodeSet as a key in a HashMap or HashSet seems weird.

However, I think there's no downside for code size to implementing Hash and Eq.  If your program doesn't use those traits, then the implementation of those traits for UnicodeSet should be dropped by the compiler.

Reply:
The [spec for PPUCD](http://site.icu-project.org/design/props/ppucd) says that the "blk" and "cp" lines can represent a single code point or a range of code points.  Even the "cp" lines can be a range of code points. It made sense to reuse UnicodeSet to represent single or ranges of code points efficiently.

Originally, I thought it made sense to use HashMap<K,V> since the code point value(s) "identify" the properties of that line in PPUCD. But on second thought, maybe that structure is more cumbersome without providing extra value over a regular Vec<(K,V)>?

Reply:
`Vec<(K, V)>` is significantly smaller, both as a structure and in terms of amount of code it pulls into the binary, and if you operate on sorted data, there's no benefit to hashing over binary search.

Reply:
Please follow up on this in #454.

Note: PpucdDataProvider is primarily an offline tool, so we don't need to go out of our way to optimize for code size.